control-plane/pkg/reconciler/broker/broker.go

```diff
@@ -359,7 +365,15 @@ func (r *Reconciler) finalizeKind(ctx context.Context, broker *eventing.Broker)
 		if err := r.finalizeNonExternalBrokerTopic(broker, securityOption, topicConfig, logger); err != nil {
 			return err
 		}
+
 	}
+
+	if secret != nil {
```

Comment:
Can we put this check inside `removeFinalizerSecret`? that would make the method more robust without changing the semantic since the semantic of that method is to remove a finalizer from a secret (but if the secret is still there is nothing to remove)